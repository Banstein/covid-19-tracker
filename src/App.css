.App {
  text-align: center;
}

/* small screens */
.ul-grid {
  display: grid;
  grid-template-columns: 1fr 1fr;
  gap: 5px;
  padding-top: 5px;
}

/* ipad */
@media screen and (min-width: 768px) and (max-width: 1080px) {
  .ul-grid {
    grid-template-columns: 1fr 1fr 1fr;
  }
}

@media screen and (min-width: 992px) {
  .ul-grid {
    grid-template-columns: 1fr 1fr 1fr 1fr;
  }
}
