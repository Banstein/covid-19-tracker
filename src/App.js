import './App.css';
import { Routes, Route } from 'react-router-dom';
import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { getCountries } from './redux/countries/countries';
import CountriesContainer from './components/CountriesContainer';
import RegionsContainer from './components/RegionsContainer';
import Navigation from './components/Navigation';

function App() {
  const dispatch = useDispatch();
  useEffect(() => {
    getCountries()(dispatch);
  }, []);
  return (
    <div>
      <Navigation />
      <Routes>
        <Route path="/" element={<CountriesContainer />} />
        <Route path="/:id" element={<RegionsContainer />} />
      </Routes>
    </div>

  );
}

export default App;
