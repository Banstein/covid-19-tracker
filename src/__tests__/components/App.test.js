import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../__mocks__/configureStore';
import App from '../../App';

describe('Countries snapshot', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <App />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
