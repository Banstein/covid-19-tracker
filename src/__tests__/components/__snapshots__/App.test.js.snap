// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries snapshot renders correctly 1`] = `
<div>
  <header
    className="navbar bg-secondary"
  >
    <ul
      className="flex justify-between container"
    >
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          <i
            className="fa-solid fa-angle-left"
          />
        </a>
      </li>
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Covid-19 Tracker
        </a>
      </li>
      <li>
        <i
          className="fa-solid fa-microphone"
        />
        <i
          className="fa-solid fa-gear"
        />
      </li>
    </ul>
  </header>
  <div>
    <div
      className="flex justify-evenly header-countries container"
    >
      <img
        alt=""
        className="world"
        src="https://simplemaps.com/static/demos/resources/svg-library/svgs/world.svg"
      />
      <h1
        className="title"
      >
        WORLD
      </h1>
    </div>
    <div
      className="bg-secondary"
    >
      <div
        className="flex container stats justify-between"
      >
        <h5
          className="title-stats"
        >
          Stats by country
           
          2022-04-15
        </h5>
        <input
          className="search-input"
          id="search"
          name="search"
          onChange={[Function]}
          placeholder="Search Countries"
          type="text"
          value=""
        />
      </div>
    </div>
    <ul
      className="ul-grid container"
    />
  </div>
</div>
`;
